validity/src/Data/Validity.hs

```diff
@@ -219,23 +164,24 @@ trivialValidation a = seq a mempty
 
 -- | Implement 'validate' by using 'isValid' and using the given string as the
 -- reason if the value is invalid.
-validateByChecking :: Validity a => String -> a -> Validation
-validateByChecking s a = isValid a <?@> s
+validateByChecking :: forall a. (a -> Bool) -> String -> a -> Validation
+validateByChecking isValid s a = isValid a <?@> s
 
 -- | Implement 'validate' by using 'isValid' and using the given name to define
 -- the reason if the value is invalid.
 --
 -- > validateByCheckingName name = validateByChecking $ unwords ["The", name, "valid."]
-validateByCheckingName :: Validity a => String -> a -> Validation
-validateByCheckingName name =
-    validateByChecking $ unwords ["The", name, "valid."]
+validateByCheckingName :: (a -> Bool) -> String -> a -> Validation
```

Comment:
I think it may be better to just remove these `validateByCheckingX` functions.
What do you think?